### ISSUE TITLE
How  to train  my model?

### ISSUE THREAD

**Issue by wuqiangch** [cunjian/face_alignment#2]
1. Can you share you code for preparing the train sample?
2. When testing an image, you only use the mean and the std of the image self, not all the mean and the std of all train image.
3. when train a model, how to deal with the sample? the mean and the std?

**Comment by cunjian** [comment]
Hi, 

This is still a piece of work in progress. It would be released once the relevant work is published. Regarding the speed, it is fairly fast since the used network is sort of shallow compared to AlexNet and VGGNet. I have not tested it with FPS on video, but you can test it by yourself. There are different ways to subtract the mean. Some work perform the subtraction with the entire training set and other work perform the subtraction on a single image. I do not observe big difference with these two methods.

**Comment by wuqiangch** [comment]
@cunjian When you  train your model, the pretreatment of training images is the same as VanillaCNN (for five landmarks)?

**Comment by cunjian** [comment]
Nope, they are sort of different. I developed my own set of pertaining schemes.

**Comment by wuqiangch** [comment]
@cunjian  where can I  find the pertaining schemes?

**Comment by wuqiangch** [comment]
@cunjian  How  to deal with the images which dlib can't find a face. Some landmarks are not in the face's bounding box, how to do do with it?

**Comment by vitoralbiero** [comment]
@wuqiangch You need to use another method to detect the face, like CNN.
However, the performance of the face_alignment with non frontal faces are very poor, so if you implement a  sofisticated method to detect the face just for alignment you will only waste your time.

**Comment by cunjian** [comment]
If you cannot fine face with DLIB, that means the landmark would not be provided. The landmark detection result relies on the location of detected face. Once the bounding box is given, the landmark could be calculated. I am not sure what do you mean by some landmarks are not within the bounding box. This should not happen since the "tanh+abs" layer would automatically shrink the range of the landmarks to be [0,1].

**Comment by cunjian** [comment]
@vitoralbiero This caffe model is trained on 300W dataset only. You can train multiple models for different poses.

**Comment by wuqiangch** [comment]
@cunjian You said you used dlib to detect faces. As in 300W dataset has the pts files. If landmarks don't lay in the rect of face from dlib, how to do deal with the rect and image ?  If dlib can't find a face in one image ,will you ignore the image directly?
